docs/configuration.md

```diff
@@ -3466,6 +3466,22 @@ project {
 }
 ```
 
+#### `project.layout`
+
+Allows specifying a project structure naming convention which can be used to
+select an appropriate dialect for cross-building if one is explicitly selected
+via [`fileOverride`](#fileoverride). By default, it's disabled (`null`).
+
+Currently, the following options are supported:
+
+- (since v3.2.0) `StandardConvention`: this is the usual naming convention
+  putting scala source code under `src/main/scala` or `src/test/scala`, with
+  alternate cross-build dialects in `src/main/scala-2.13`
+
+If set, for minor versions, dialect will be determined automatically, as if with
+`fileOverride { "lang:scala-2.13" = scala213 }`. Currently, supports
+scala versions 2.10 through 2.13.
```

Comment:
scala-3 is a major version. we can't deterministically figure out which specific minor dialect to use, hence automatic dialect is only for minor versions.

are there scala 3.0x already in active use?

Reply:
We started using that convention here: https://github.com/scalameta/metals/tree/main/mtags/src/main/scala-3/scala/meta/internal

It should be less common for users to use `scala-3.0` or `scala-3.1`, but I suppose it could happen. However I don't expect the convention to be different that in Scala 2. 

We could default for now to dialect `Scala3` for now I think. We will need to update it anyway when some drastic things happen, though `3.1` didn't change anything syntactic still.

Reply:
do you know how sbt cross build determines source directory to use for a given scala version by default? that's the only thing we should realistically support here.

for metals, i found this complex, non-default logic: https://github.com/scalameta/metals/blob/main/build.sbt

Reply:
specifically, I'm looking for scala3 version-specific subdirs like https://www.scala-sbt.org/1.x/docs/Cross-Build.html#Scala-version+specific+source+directory

Reply:
I checked with sbt and it detects `scala-3` as a Scala 3 folder by default. It doesn't happen with `3.0` and `3.1`. On the other hand, for Scala 2 it works both with `scala-2` and with `scala-2.13`.

Reply:
you mean, not just with mtags but any default cross build?

also, if it works, the difference is likely this: sbt knows the dialect it's building, and just needs to find the source directories for it. however, scalafmt is opposite: knows the directory but doesn't know the dialect.

it's easy to determine if from scala-2.13; what should it be for scala-3?

and for this logic (https://www.scala-sbt.org/1.x/docs/Cross-Build.html#Scala-version+specific+source+directory), how will they apply it to scala 3? would it expect scala-3.10 for 3.1.0, or scala-3.1 (since the migration guide says scala3 uses major.minor.patch instead of epoch.major.minir for scala2).

Reply:
ok, found this: https://github.com/sbt/sbt/pull/6424

above the changes, lines 745 and 760, they are reading sources from `scalaBinaryVersion`. i don't know if default `scalaBinaryVersion` for dotty is `scala-3` or `scala-3.1` or `scala-3.1.0`.

i think right now we can do one of two things:
- add dialect support for `scala-3.0` and `scala-3.1` as that would allow us to determine the dialect reasonably
- revisit this case later, when we have a better understanding.

what would you prefer, @tgodzik?

Reply:
I checked the default cross build sbt behaviour it recognizes `scala-3` directory by default in a sample project. It doesn't work for `scala-3.0` etc.

The behaviour in https://github.com/sbt/sbt/pull/6424 was for release candidates that no longer applies. 

I think it's safe to assume that `scala-3` will always work. Having `3.0` and `3.1` directories is no in the plans AFAIK

Reply:
ok, done. added `scala-3`.

Reply:
Thanks!